### PR TITLE
Updating the Invoice Date logic

### PR DESCRIPTION
**Pull request by Vlad-CL**
# JIRA
https://clxmedia.atlassian.net/browse/CLXD-1754

# Description

Now Invoice Date could be changed after invoice posting and bulk Invoice Change action added.

### CODE REVIEW

File: clx_invoice_policy/models/account_move_invoicedate.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from odoo.exceptions import UserError
+
+
+class AccountMoveInvoiceDate(models.TransientModel):
+    _name = "account.move.invoicedate"
+    invoice_date = fields.Date(string="New Invooice Date")
+
+    def set_invoice_date(self):
+        if self._context.get("active_model") != "account.move":
+            raise UserError(_("The invoice items coulds be updated only."))
```
Comment (CaseyDStein) [inline clx_invoice_policy/models/account_move_invoicedate.py:11]: Please change this error message to say something like: "We were unable to move the invoice date.  Only the invoice items could be updated."

File: clx_invoice_policy/models/account_move_invoicedate.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class AccountMoveInvoiceDate(models.TransientModel):
+    _name = "account.move.invoicedate"
+    invoice_date = fields.Date(string="New Invooice Date")
```
Comment (CaseyDStein) [inline clx_invoice_policy/models/account_move_invoicedate.py:7]: Slight misspelling: "Invooice" should be "Invoice"

File: clx_invoice_policy/models/account_move_invoicedate.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        if self._context.get("active_model") != "account.move":
+            raise UserError(_("The invoice items coulds be updated only."))
+
+        if not self.invoice_date:
+            raise UserError(_("The new Invoice Date is not correct."))
+
+        domain = [("id", "in", self._context.get("active_ids", [])), ("state", "=", "draft")]
+        moves = self.env["account.move"].search(domain).filtered("line_ids")
+        if not moves:
+            raise UserError(_("There are no invoices selectet to be updated."))
```
Comment (CaseyDStein) [inline clx_invoice_policy/models/account_move_invoicedate.py:19]: Slight misspelling: "selectet" should be "selected"

**Review by CaseyDStein — CHANGES_REQUESTED** [review]
This looks good.  I have a couple suggestions related to spelling and wording.

File: clx_invoice_policy/models/account_move_invoicedate.py (hunk 1 above)
Comment (Vlad-CL) [reply clx_invoice_policy/models/account_move_invoicedate.py:11]: Done

File: clx_invoice_policy/models/account_move_invoicedate.py (hunk 2 above)
Comment (Vlad-CL) [reply clx_invoice_policy/models/account_move_invoicedate.py:7]: Thanks, good catch

File: clx_invoice_policy/models/account_move_invoicedate.py (hunk 3 above)
Comment (Vlad-CL) [reply clx_invoice_policy/models/account_move_invoicedate.py:19]: Thanks

**Review by CaseyDStein — APPROVED** [review]
Looks great!  Thanks for making those changes!

**Pull request merged**